File: frontend/src/types/cytoWorkload.ts
export interface CytoWorkloadDayStats {
  user_id: number;
  user_full_name: string;
  work_date: string;
  gyne_slides: number;
  nongyne_conv_slides: number;
  nongyne_liquid_slides: number;
  effective_count: number;
  reading_hours: number | null;
  adjusted_limit: number;
  is_compliant: boolean;
  note?: string;
}

export interface CytoWorkloadLogUpsert {
  user_id: number;
  work_date: string;
  reading_hours: number;
  note?: string;
}

export interface CytoWorkloadLogResponse {
  id: number;
  user_id: number;
  work_date: string;
  reading_hours: number;
  note?: string;
  recorded_by_id?: number;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/cyto_workload.py
from pydantic import ConfigDict, BaseModel, field_validator
from datetime import date, datetime
from typing import Optional


class CytoWorkloadLogUpsert(BaseModel):
    user_id: int
    work_date: date
    reading_hours: float
    note: Optional[str] = None

    @field_validator("reading_hours")
    @classmethod
    def hours_positive(cls, v: float) -> float:
        if v < 0:
            raise ValueError("reading_hours must be >= 0")
        return v


class CytoWorkloadLogResponse(BaseModel):
    id: int
    user_id: int
    work_date: date
    reading_hours: float
    note: Optional[str] = None
    recorded_by_id: Optional[int] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class CytoWorkloadDayStats(BaseModel):
    user_id: int
    user_full_name: str
    work_date: date
    gyne_slides: int
    nongyne_conv_slides: int
    nongyne_liquid_slides: int
    effective_count: float
    reading_hours: Optional[float]
    adjusted_limit: float
    is_compliant: bool
    note: Optional[str] = None
